TLM/TLM/Manager/Impl/LaneConnectionManager.cs

```diff
@@ -80,48 +102,44 @@ public class LaneConnectionManager
             throw new Exception($"Unreachable code. sourceLaneId:{sourceLaneId}, segmentId:{segmentId} ");
         }
 
-        public bool HasConnections(uint sourceLaneId) {
-            if (!Options.laneConnectorEnabled) {
-                return false;
-            }
-            return HasConnections(sourceLaneId, IsHeadingTowardsStartNode(sourceLaneId));
-        }
-
-        public bool HasConnections(uint sourceLaneId, bool startNode) {
+        public bool HasOutgoingConnections(uint sourceLaneId) {
             if (!Options.laneConnectorEnabled) {
                 return false;
             }
 
-            int nodeArrayIndex = startNode ? 0 : 1;
-
-            return Flags.laneConnections[sourceLaneId] != null &&
-                   Flags.laneConnections[sourceLaneId][nodeArrayIndex] != null;
+            return HasOutgoingConnections(sourceLaneId, IsHeadingTowardsStartNode(sourceLaneId));
```

Comment:
Although not touched by this PR yet, there might be some issues with `IsHeadingTowardsStartNode()` to consider as it checks specifically for `NetInfo.Direction.Forward` (taking in to account inverted etc).

- What happens if direction is `Both`
- Or any of the new `Avoid` things? - as detailed in https://github.com/CitiesSkylinesMods/TMPE/issues/1337#issuecomment-1025008063

Also, that method uses `foreach` which is about 2x slower than `for` loop IIRC?

Reply:
its only used by lane arrow manager.

Reply:
its not performance critical

Reply:
I would leave it like it is. Issue with Avoid flags was before and normal roads where user can manager lane connections are not available so I'm not sure if it's worth to improve that in this PR. IMO a bit out of scope